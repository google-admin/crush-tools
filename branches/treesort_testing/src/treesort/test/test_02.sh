test_number=02
description="three column sort"

subtest_desc=("CC,BB,DD" 
							"AA,BB,CC"
							"DD,CC,BB"
							"4,2,1 with indexes"
							'^AA,BB,CC'
							'^AA,^BB,CC'
							'^AA,^BB,^CC'
							'1,^2,^3 with indexes'
							'AA,BB,^CC'
							'1,^2,3 with indexes'
							)
subtest_opts=(
'-K CC,BB,DD'
'-K AA,BB,CC'
'-K DD,CC,BB'
'-k 4,2,1 -p'
'-K ^AA,BB,CC'
'-K ^AA,^BB,CC'
'-K ^AA,^BB,^CC'
'-k 1,^2,^3 -p'
'-K AA,BB,^CC'
'-k 1,^2,3 -p'
)

for subtest in `seq 0 9`; do
  expected="$test_dir/test_$test_number.$subtest.expected"
  outfile="$test_dir/test_$test_number.$subtest.actual"

  $bin ${subtest_opts[$subtest]} \
       "$test_dir/test2.in" \
       > "$outfile"

  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" FAIL
  else
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" PASS
    rm "$outfile"
  fi
done
