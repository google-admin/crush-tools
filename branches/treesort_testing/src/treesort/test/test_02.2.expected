DDþCCþBBþAA
11.9550þ2009-10-09þZeP1N6ZHo8þ222
11.9550þ2009-10-15þOxNCXf_QO5þ222
11.9550þ2009-10-15þV-dWH8XhzNþ704
11.9550þ2009-10-18þoj-npio3Cqþ119
11.9550þ2009-10-21þPb0jUNXPGFþ704
11.9550þ2009-10-21þlAkjpi7B0uþ704
11.9550þ2009-10-21þqdigsUU1EHþ222
11.9550þ2009-10-29þ5-dmH-yGymþ120
13.3454þ2009-10-15þOypnZVbJOYþ876
13.3454þ2009-10-15þbNfKeDJp35þ120
13.3454þ2009-10-18þdCzebYQfiHþ222
13.3454þ2009-10-18þmdg21XYvE4þ379
13.3454þ2009-10-19þDx5ojQn-G6þ379
13.3454þ2009-10-21þBuPJhoNfZYþ692
13.3454þ2009-10-21þD6DavOvkp5þ119
13.3454þ2009-10-23þHF90x6_QiBþ222
13.3454þ2009-10-25þgbwjbbmOR5þ704
13.3454þ2009-10-29þ_Dh-EwmwHUþ222
13.3454þ2009-10-29þeGWkG9dnHHþ790
14.6088þ2009-10-15þ1aRiJ5zvaUþ379
14.6088þ2009-10-15þQ99ceOGe58þ119
14.6088þ2009-10-18þY0_LegXodZþ120
14.6088þ2009-10-18þqv6ulolvVmþ876
14.6088þ2009-10-19þQHJtym0emOþ790
14.6088þ2009-10-21þQEIlvFBOAEþ428
14.6088þ2009-10-21þooYbIsNctcþ692
16.2961þ2009-10-02þvT9F16VWxuþ876
16.2961þ2009-10-06þRHEIKKC-eNþ876
16.2961þ2009-10-06þw-DsqhmkP4þ428
16.2961þ2009-10-15þU_DsBKeZBOþ790
16.2961þ2009-10-18þ3Abgi5FvpKþ692
16.2961þ2009-10-18þFJFwWbUCEgþ525
16.2961þ2009-10-23þab_ALFxdBlþ704
16.2961þ2009-10-25þGkijsWrtDmþ222
16.2961þ2009-10-29þAnKSv96_ykþ525
16.2961þ2009-10-29þJw-SmuJJwHþ428
16.2961þ2009-10-29þXdQde3c2LRþ120
16.2961þ2009-10-29þkX69CHRCpFþ119
19.8086þ2009-10-02þMP3BenV_PWþ525
19.8086þ2009-10-02þcOA-GflWf9þ120
19.8086þ2009-10-09þV_Srbg1Xgkþ525
19.8086þ2009-10-15þqHbD7IgvBsþ119
19.8086þ2009-10-19þ0J2H9eMxz5þ428
19.8086þ2009-10-19þ1Qbw9NbEK2þ120
19.8086þ2009-10-21þMk8GiTvT-2þ222
19.8086þ2009-10-23þWxgda_Rz1uþ120
19.8086þ2009-10-25þN1CuDuqLsqþ120
19.8086þ2009-10-29þCwEfrzVzlNþ704
20.0623þ2009-10-15þtH_urSAL6Iþ379
20.0623þ2009-10-18þQOhQYF2L4Uþ120
20.0623þ2009-10-18þR0lHfvqeE2þ222
20.0623þ2009-10-19þQddzqzZhvaþ222
20.0623þ2009-10-19þs5FX4ERZP8þ119
20.0623þ2009-10-21þUH3YCtEsnEþ379
20.0623þ2009-10-23þgplXXJQCbDþ692
20.0623þ2009-10-25þSEiwhaVJcvþ790
20.0623þ2009-10-25þi4Vn-mjUX_þ379
20.0623þ2009-10-29þH3laCVXVqFþ525
21.4784þ2009-10-09þDMqu6KjLMnþ428
21.4784þ2009-10-09þZ_RNo9lPThþ222
21.4784þ2009-10-09þ_6GXpFrx7Mþ704
21.4784þ2009-10-15þ-_WgFpelwTþ119
21.4784þ2009-10-19þ-l43yW3QQXþ379
21.4784þ2009-10-19þQOa1zDDXrcþ692
21.4784þ2009-10-19þZs9uH9wqwvþ120
21.4784þ2009-10-19þ_okpFO60P5þ120
21.4784þ2009-10-19þnLKFOKtBt5þ876
21.4784þ2009-10-19þyj8FAR6sT8þ790
21.4784þ2009-10-23þ6RIZ_xP77Oþ876
21.4784þ2009-10-25þK26asHqM1Bþ692
21.4784þ2009-10-25þMLySuT_6L-þ119
21.4784þ2009-10-25þghqsjQH7CAþ790
21.4784þ2009-10-25þkC3oWoj1_8þ692
21.4784þ2009-10-29þpfetgY4yYtþ525
23.1071þ2009-10-02þS-BUVAIarsþ120
23.1071þ2009-10-02þcEXl50cdsGþ790
23.1071þ2009-10-06þVjx-O1uUiyþ119
23.1071þ2009-10-06þbWuILAzu5Lþ379
23.1071þ2009-10-09þNrNOPtKMh2þ222
23.1071þ2009-10-18þi0-lOQWHXeþ428
23.1071þ2009-10-19þpPrA9-ibvPþ790
23.1071þ2009-10-19þuGBjnSxQwIþ692
23.1071þ2009-10-21þZ971TGdirGþ876
23.1071þ2009-10-21þj-OdwoZTzmþ692
23.1071þ2009-10-29þ_mNRE6q64Dþ525
25.4415þ2009-10-09þOcgRMuIUVWþ120
25.4415þ2009-10-15þccfVWBl1Ifþ379
25.4415þ2009-10-18þczHP0Y09gOþ119
25.4415þ2009-10-19þ8ZDolJWFhKþ120
25.4415þ2009-10-19þNqFghFb16Pþ119
25.4415þ2009-10-21þ0CjQNzoprAþ790
25.4415þ2009-10-23þQg_963dGw2þ428
25.4415þ2009-10-25þn1pETdd_e2þ790
43.1227þ2009-10-02þ8HPzHGZLDfþ222
43.1227þ2009-10-06þPKndexdB5Yþ119
43.1227þ2009-10-09þl0sIDKi1Mnþ222
43.1227þ2009-10-15þpUbcEzSLg5þ379
43.1227þ2009-10-18þ-ynhLNQjXnþ525
43.1227þ2009-10-19þ-ca4VazCCoþ525
43.1227þ2009-10-21þzHaJcq1brbþ379
